test/unit/safari_test.rb

```diff
@@ -66,6 +66,31 @@ class SafariTest < Minitest::Test
     assert_equal "9.0.2", browser.full_version
   end
 
+  test "reject other iOS and non-Safari browsers" do
+    %w[
+      ALIPAY_IOS
+      CHROME
+      DUCKDUCKGO_BROWSER_IOS
+      FIREFOX_IOS
+      GOOGLE_SEARCH_APP_IPHONE
+      HUAWEI_BROWSER
+      MAXTHON
+      MIUI_BROWSER
+      MS_EDGE
+      OPERA
+      PHANTOM_JS
+      QQ_BROWSER_IOS
+      SAMSUNG_BROWSER
+      SOUGOU_BROWSER
+      SPUTNIK
+      UC_BROWSER
+      YANDEX_BROWSER_IOS
+    ].each do |user_agent|
+      browser = Browser.new(Browser[user_agent])
+      refute browser.safari?
+    end
+  end
```

Comment:
This is already being tested by each test case (or at least they should); see [alipay_test.rb](https://github.com/fnando/browser/blob/main/test/unit/alipay_test.rb#L14-L15) for example. Can you add assertions for this to uc_browser_test.rb and other missing browsers from this list?

Reply:
As unit tests, shouldn’t these be testing the behavior of `safari.rb` though? `safari.rb` is where the extra match logic lives for rejecting other browsers. 

Currently it’s not clear that if you update `safari.rb` you would need to update tests for unrelated browsers.

Reply:
Like you mentioned, “at least they should”, it’s much harder to verify we’re testing against all those cases if they live in other files.